core/vcpu.c

```diff
@@ -388,6 +390,19 @@ static int (*handler_funcs[])(struct vcpu_t *vcpu, struct hax_tunnel *htun) = {
     [VMX_EXIT_FAILED_VMENTER_GS]  = exit_invalid_guest_state,
     [VMX_EXIT_EPT_VIOLATION]      = exit_ept_violation,
     [VMX_EXIT_EPT_MISCONFIG]      = exit_ept_misconfiguration,
+    [VMX_EXIT_GETSEC]             = exit_unsupported_instruction,
+    [VMX_EXIT_INVD]               = exit_unsupported_instruction,
+    [VMX_EXIT_VMCALL]             = exit_unsupported_instruction,
+    [VMX_EXIT_VMCLEAR]            = exit_unsupported_instruction,
+    [VMX_EXIT_VMLAUNCH]           = exit_unsupported_instruction,
+    [VMX_EXIT_VMPTRLD]            = exit_unsupported_instruction,
+    [VMX_EXIT_VMPTRST]            = exit_unsupported_instruction,
+    [VMX_EXIT_VMREAD]             = exit_unsupported_instruction,
```

Comment:
Why VMREAD/WMWRITE registered to unconditional exit handler?

Reply:
Cpuid handling in hax tells the guest that it does not support VT. So execution of any instruction in this subset should generate an #UD. Intel SDM says that exits of these instructions are conditionally, the conditions in hax lead to exits. I tested these instructions and they exited.

Reply:
Maybe it makes the code more clear by adding comments after each of these vmx_exit_* item, like already did in the enum, tells that some are due to unconditional exit while other are conditional according to the SDM 3C 25.1.2.

Reply:
Added comments.